vaadin-text-field-flow-parent/vaadin-text-field-testbench/src/main/java/com/vaadin/flow/component/textfield/testbench/BigDecimalFieldElement.java

```diff
@@ -32,11 +32,21 @@
 public class BigDecimalFieldElement extends TestBenchElement
         implements HasStringValueProperty, HasLabel, HasPlaceholder, HasHelper {
 
+    /**
+     * Emulates the user changing the value, which in practice means setting
+     * {@code value} of the {@code input} element to the given value and then
+     * triggering {@code input} and {@code change} DOM events.
+     *
+     * @param string
+     *            the value to set
+     */
     @Override
     public void setValue(String string) {
-        HasStringValueProperty.super.setValue(string);
-        dispatchEvent("change", Collections.singletonMap("bubbles", true));
-        dispatchEvent("blur");
+        TestBenchElement input = $("input").first();
+        input.setProperty("value", string);
+        input.dispatchEvent("input", Collections.singletonMap("bubbles", true));
+        input.dispatchEvent("change",
+                Collections.singletonMap("bubbles", true));
```

Comment:
Nit: We could consider extracting this in a static helper that receives the TestBench element. Don't have a strong opinion on this though, the code is rather simple.

Reply:
Do you have an idea of possible naming for such a helper class?

Reply:
We can make it package-private, so naming isn't super important as it can be changed anytime. I would go with a more general `TextFieldElementHelper`, but your suggestion is fine too.

Reply:
👍  Done.